From accec2e7109e33b482b9339125c22f523b6c1a66 Mon Sep 17 00:00:00 2001
From: Marty Jones <mj8263788@gmail.com>
Date: Sun, 29 Nov 2020 02:31:31 -0500
Subject: [PATCH] wireguard: fix compile with kernel 5.10

Signed-off-by: Marty Jones <mj8263788@gmail.com>
---
 src/compat/compat.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -34,6 +34,7 @@
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 1, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
 #define ISUBUNTU1904
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
 #define ISUBUNTU1910
 #endif
 #endif
@@ -42,8 +43,8 @@
 #error "WireGuard requires Linux >= 3.10"
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
-#error "WireGuard has been merged into Linux >= 5.6 and therefore this compatibility module is no longer required."
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
+#error "WireGuard has been merged into Linux >= 5.6 and therefore this compatibility module is no longer required."
 #endif
 
 #if defined(ISRHEL7)
